### PR TITLE
Update CP1250 font support - Slovak characters

### PR DESCRIPTION
**Pull request by Roxterat**
Update of CP1250 font support for characters:
ľ, ĺ, ô, ŕ, ä (both uppercase and lowercase, normal and small font, extra character "â" added too).
ť (lowercase t with caron) fixed.

### CODE REVIEW

**Comment by Roxterat** [comment]
Trying to figure if "ľ " can exist and what's wrong with checks..

**Comment by Roxterat** [comment]
Checks done, will see about lowercase L with caron.

**Comment by Roxterat** [comment]
Lowercase L with caron solved, opening PR for review @zenseii .

**Comment by Roxterat** [comment]
Oh, and also @Districh-ru , your review would be much appreciated as well, sorry I have mentioned only Zense, old habids from translations. :)

File: src/fheroes2/gui/ui_font.cpp [hunk 1]
```diff
@@ -301,6 +304,31 @@ namespace
...
+            font[188 - 32].reset();
+            fheroes2::Copy( font[76 - 32], 0, 0, font[188 - 32], 0, 1, font[76 - 32].width(), font[76 - 32].height() );
+            fheroes2::Copy( font[65], 1, 1, font[188 - 32], 9, 1, 1, 1 );
+            fheroes2::Copy( font[65], 1, 1, font[188 - 32], 10, 0, 1, 1 );
+            fheroes2::Copy( font[65], 1, 1, font[188 - 32], 8, 0, 1, 1 );
+            fheroes2::Copy( font[65], 1, 0, font[188 - 32], 8, 1, 1, 1 );
+            fheroes2::Copy( font[65], 1, 0, font[188 - 32], 9, 0, 1, 1 );
+            fheroes2::Copy( font[65], 1, 0, font[188 - 32], 7, 0, 1, 1 );
+            font[188 - 32].setPosition( font[76 - 32].x(), font[76 - 32].y() - 1 );
+            updateNormalFontLetterShadow( font[188 - 32] );
```
Comment (Districh-ru) [inline src/fheroes2/gui/ui_font.cpp:318]: According to the available fonts I can notice that изображение should have its caron not above the `L` but in the upper right corner of it, like in Normal font. What do you think?

Interesting, but in modern fonts the изображение ([Ľ](https://www.ascii-code.com/character/%C4%BD)) character is represented with apostrophe. :)
It is up to you @Roxterat what variant is better.

**Review by Districh-ru — CHANGES_REQUESTED** [review]
Well done, @Roxterat!
I have one question, could you take a look when you have time.

Also here is a screenshot from the game for letters from CP1250:
Š Ś Ť Ž Ź š ś ť ž ź Ł Ą Ş Ż ł µ ą ş Ľ ľ ż Ŕ Á Â Ă Ä Ĺ Ć Ç Č É Ę Ë Ě Í Î Ď Đ Ń Ň Ó Ô Ő Ö Ř Ů Ú Ű Ü Ý Ţ ß ŕ á â ă ä ĺ ć ç č é ę ë ě í î ď đ ń ň ó ô ő ö ř ů ú ű ü ý ţ 

изображение

Not to forget in future - there are still some letters to add to CP1250: [Ş](https://www.ascii-code.com/character/%C5%9E) [µ](https://www.ascii-code.com/character/%C2%B5) [Ă](https://www.ascii-code.com/character/%C4%82) [Ç](https://www.ascii-code.com/character/%C3%87) [Ë](https://www.ascii-code.com/character/%C3%8B) [Î](https://www.ascii-code.com/character/%C3%8E) [Đ](https://www.ascii-code.com/character/%C4%90) [Ţ](https://www.ascii-code.com/character/%C5%A2) [ß](https://www.ascii-code.com/character/%C3%9F) [ă](https://www.ascii-code.com/character/%C4%83) [ç](https://www.ascii-code.com/character/%C3%A7) [ë](https://www.ascii-code.com/character/%C3%AB) [î](https://www.ascii-code.com/character/%C3%AE) [đ](https://www.ascii-code.com/character/%C4%91) [ţ](https://www.ascii-code.com/character/%C5%A3)
They can be done later in a new PR when it will be needed. I don't request to add them in current PR. :)

**Comment by Districh-ru** [comment]
> Thank you for the better visuals, I shall attempt to present the work in PRs in better quality in further contributions myself. Yes, there are still characters that needs doing, I can safely make some of them but I am not completely certain about how all should be properly made, I might attempt in future whatsoever.

@Roxterat, I just replaced the translation for save file info in Polish translation file with all these letters. And now when I click the right mouse button on a save in load dialog I can see them rendered by the game engine. Here is the changed .mo file, you can use it in your tests: [pl.zip](https://github.com/ihhub/fheroes2/files/11480499/pl.zip)

**Comment by Roxterat** [comment]
> > Thank you for the better visuals, I shall attempt to present the work in PRs in better quality in further contributions myself. Yes, there are still characters that needs doing, I can safely make some of them but I am not completely certain about how all should be properly made, I might attempt in future whatsoever.
> 
> @Roxterat, I just replaced the translation for save file info in Polish translation file with all these letters. And now when I click the right mouse button on a save in load dialog I can see them rendered by the game engine. Here is the changed .mo file, you can use it in your tests: [pl.zip](https://github.com/ihhub/fheroes2/files/11480499/pl.zip)

Nice, thanks for a set file for testing!

**Comment by Roxterat** [comment]
@Districh-ru and @zenseii , commited fixes, charcters should be resolved accordingly.


156s



line for Lowercase a with circumflex:
fheroes2::Copy( font[194 - 32], 7, 0, font[226 - 32], 3, 0, 3, 2 );

File: src/fheroes2/gui/ui_font.cpp [hunk 2]
```diff
@@ -477,6 +564,35 @@ namespace
...
 
+            // Lowercase a with circumflex
+            font[226 - 32].resize( font[97 - 32].width(), font[97 - 32].height() + 3 );
+            font[226 - 32].reset();
+            fheroes2::Copy( font[97 - 32], 0, 0, font[226 - 32], 0, 3, font[97 - 32].width(), font[97 - 32].height() );
+            fheroes2::Copy( font[97 - 32], 7, 1, font[226 - 32], 3, 1, 1, 1 );
+            fheroes2::Copy( font[97 - 32], 7, 1, font[226 - 32], 4, 0, 1, 1 );
+            fheroes2::Copy( font[97 - 32], 7, 1, font[226 - 32], 5, 1, 1, 1 );
+            fheroes2::Copy( font[97 - 32], 1, 0, font[226 - 32], 3, 0, 1, 1 );
+            fheroes2::Copy( font[97 - 32], 1, 0, font[226 - 32], 5, 0, 1, 1 );
```
[the remark this answers is not in the document]
Comment (Roxterat) [reply src/fheroes2/gui/ui_font.cpp]: Solution in comments.

File: src/fheroes2/gui/ui_font.cpp (hunk 1 above)
Comment (Roxterat) [reply src/fheroes2/gui/ui_font.cpp]: Solution in comments.

**Review by Districh-ru — APPROVED** [review]
👍

**Review by oleg-derevenetz — APPROVED** [review]

**Comment by oleg-derevenetz** [comment]
@Roxterat thank you!

**Pull request merged**